diplomacy/engine/game.py

```diff
@@ -978,10 +979,21 @@ def filter_messages(cls, messages, game_role, timestamp_from=None, timestamp_to=
             game_role = [game_role]
         elif not isinstance(game_role, list):
             game_role = list(game_role)
+
+        def is_advice_for_role(message: Message) -> bool:
+            """Return True if this message is advice for given game role(s)."""
+            return (
+                message.type in strings.ALL_SUGGESTION_TYPES
+                and json.loads(message.message).get("recipient") in game_role
```

Comment:
This is fine for message content that contains json payload. Can you add a try catch that prevents `JSONDecodeError` though?

Reply:
I can make the code more robust (and definitely considered it), but it'd make the code a lot more complicated.

I don't think it's needed because there should never be a suggestion message with invalid JSON, and operator short-circuiting will prevent attempting to parse JSON from a non-suggestion message.

I can definitely make the code more resilient if you think that's important, though.